addons/account/models/account_account.py

```diff
@@ -506,12 +506,19 @@ def default_get(self, default_fields):
         return super(AccountAccount, contextual_self).default_get(default_fields)
 
     @api.model
-    def _order_by_frequency_per_partner(self, company_id, partner_id, move_type=None, limit=None):
+    def _get_most_frequent_accounts_for_partner(self, company_id, partner_id, move_type, join='left join', limit=None):
```

Comment:
Don't pass directly how you want to join your sql query, I prefer a parameter like `filter_accounts=True/False`
And in your code, you do your join according to this parameter